### ISSUE TITLE
jsonAttributes causing errors on server start

### ISSUE THREAD

**Issue by wswoodruff** [hapipal/schwifty#18]
Where:
This is happening when trying to start the server with schwifty's master branch, and the `add-schwifty` branch of user-boilerplate: https://github.com/WilliamSWoodruff/user-boilerplate/tree/add-schwifty.

Repro:
* You will `not` need a working db username, login, or db name, this error shows up before that.
* Clone both repos, npm link schwifty into user-boilerplate, run npm install on both and you'll see this stacktrace

Stacktrace:
```
/Users/myUser/Developer/schwifty/node_modules/objection/lib/model/Model.js:809
      this.jsonAttributes = [];
                          ^

TypeError: Cannot set property jsonAttributes of class SchwiftyModel extends Objection.Model {

    static get joiSchema() {}

    // Caches schema, with and wi...<omitted>...
} which has only a getter
    at Function.getJsonAttributes (/Users/myUser/Developer/schwifty/node_modules/objection/lib/model/Model.js:809:27)
    at Tokens.$parseDatabaseJson (/Users/myUser/Developer/schwifty/node_modules/objection/lib/model/Model.js:297:37)
    at Function.columnNameToPropertyName (/Users/myUser/Developer/schwifty/node_modules/objection/lib/model/ModelBase.js:601:50)
    at Function.decorator$memoize [as columnNameToPropertyName] (/Users/myUser/Developer/schwifty/node_modules/objection/lib/utils/decorators/memoize.js:63:24)
    at /Users/myUser/Developer/schwifty/node_modules/objection/lib/relations/Relation.js:565:37
    at Array.map (native)
    at BelongsToOneRelation.propertyName (/Users/myUser/Developer/schwifty/node_modules/objection/lib/relations/Relation.js:564:20)
    at BelongsToOneRelation.setMapping (/Users/myUser/Developer/schwifty/node_modules/objection/lib/relations/Relation.js:272:27)
    at /Users/myUser/Developer/schwifty/node_modules/objection/lib/model/Model.js:712:33
    at /Users/myUser/Developer/schwifty/node_modules/lodash/lodash.js:935:11
    at /Users/myUser/Developer/schwifty/node_modules/lodash/lodash.js:4944:15
    at baseForOwn (/Users/myUser/Developer/schwifty/node_modules/lodash/lodash.js:3001:24)
    at /Users/myUser/Developer/schwifty/node_modules/lodash/lodash.js:4913:18
    at baseReduce (/Users/myUser/Developer/schwifty/node_modules/lodash/lodash.js:932:5)
    at Function.reduce (/Users/myUser/Developer/schwifty/node_modules/lodash/lodash.js:9698:14)
    at Function.getRelations (/Users/myUser/Developer/schwifty/node_modules/objection/lib/model/Model.js:710:36)
    at Function.bindKnex (/Users/myUser/Developer/schwifty/node_modules/objection/lib/model/Model.js:567:32)
    at models.forEach (/Users/myUser/Developer/schwifty/lib/index.js:107:56)
    at Array.forEach (native)
    at collector.knexGroups.forEach (/Users/myUser/Developer/schwifty/lib/index.js:103:16)
    at Array.forEach (native)
    at internals.initialize (/Users/myUser/Developer/schwifty/lib/index.js:98:26)
```

**Comment by devinivy** [comment]
This occurs because we define `jsonAttributes` as a getter but don't have a corresponding setter.  We can either sidestep the case that objection sets `jsonAttributes` itself (never compute `jsonAtrributes` as `null`– instead `[]`), or add a setter.

**Comment by wswoodruff** [comment]
Howbout we just make a blatant setter so nobody ever wonders why it's `[]` by default, most ppl really won't know why we're putting it there though loL

**Comment by mattboutet** [comment]
Seems like a setter with a comment that explains why it's there would do the trick.

**Comment by devinivy** [comment]
I think I'd opt to instead nix the `jsonAttributes` getter altogether, and instead extend the `getJsonAttributes` static method.  I'm wishing I had done that from the start now.

**Comment by devinivy** [comment]
Ah, it's hard.  It's "protected" but not private-prefixed (mixed signs), but it's also where the implemented the same functionality for jsonSchema (good).  Hrmmm!

https://github.com/Vincit/objection.js/blob/master/src/model/Model.js#L678-L682

**Comment by wswoodruff** [comment]
Creating the setter is the best option to avoid any breaking updates from Objection right?

**Comment by wswoodruff** [comment]
I got the server to start with this code adding a setter, if we choose that method: https://github.com/BigRoomStudios/schwifty/pull/19

**Comment by devinivy** [comment]
It only matters if objection will stop using `getJsonAttributes()` in the future or its meaning changes.  I would really love to put the logic to compute the `jsonAttributes` there– it's much cleaner.  Maybe we should just ask!

@koskimas Hi– we have an objection.js question for you!  We're in the process of implementing default `jsonAttributes` for the joi validation library, as opposed to json-schema.  We see that the default `jsonAttributes` are computed from a json-schema in `Model.getJsonAttributes()`, and ideally I think that's where we would put our implementation.  However, we also see that this static method is marked as "protected."  So, can we safely override that method, or should we sidestep it by implementing `static get jsonAttributes()` instead?

**Comment by devinivy** [comment]
He got back to me on gitter,
> @devinivy Anything not documented [here](http://vincit.github.io/objection.js/) is considered private implementation. The @protected is there just to prevent Intellij from complaining about private method usage. It used to be in BaseModel. Now the annotation could actually be removed. So you are better off implementing jsonAttributes as a getter if that works for you?

getter/setter it is!

**Comment by wswoodruff** [comment]
Hurrah, an answer! Good idea to ask, thanx!

**Comment by devinivy** [comment]
Resolved by #19
